raft/raft.go

```diff
@@ -172,6 +173,9 @@ func newRaft(id uint64, peers []uint64, election, heartbeat int, storage Storage
 	if !isHardStateEqual(hs, emptyState) {
 		r.loadState(hs)
 	}
+	if applied > 0 {
```

Comment:
The applied parameter as passed in, or `raftlog.applied` (which is effectively `max(applied, raftlog.firstIndex())`)?